src/commands/dev/index.js

```diff
@@ -82,7 +91,30 @@ const startFrameworkServer = async function ({ settings, log, exit }) {
   })
 
   try {
-    const open = await waitPort({ port: settings.frameworkPort, output: 'silent', timeout: FRAMEWORK_PORT_TIMEOUT })
+    const open = await waitPort({
+      port: settings.frameworkPort,
+      output: 'silent',
+      timeout: FRAMEWORK_PORT_TIMEOUT,
+    })
+
+    const waitForServerToRespond = async () => {
+      try {
+        await fetch(`http://localhost:${settings.frameworkPort}`, {
+          method: 'HEAD',
+          timeout: SERVER_POLL_REQUEST_TIMEOUT,
+        })
+      } catch (_) {
+        return false
+      }
+
+      return true
+    }
+
+    await waitFor(waitForServerToRespond, {
```

Comment:
Can we skip this `waitFor` if `open` is `false`?

Reply:
Absolutely.